nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -260,4 +285,123 @@ protected Update updateRecordToKudu(KuduTable kuduTable, Record record, List<Str
+                case INT16:
+                    jsonBuilder.append("\"" + row.getShort(col.getName()) + "\"");
+                    break;
+                case INT32:
+                    jsonBuilder.append("\"" + row.getInt(col.getName()) + "\"");
+                    break;
+                case INT64:
+                    jsonBuilder.append("\"" + row.getLong(col.getName()) + "\"");
+                    break;
+                case BOOL:
+                    jsonBuilder.append("\"" + row.getBoolean(col.getName()) + "\"");
+                    break;
+                case DECIMAL:
+                    jsonBuilder.append("\"" + row.getDecimal(col.getName()) + "\"");
+                    break;
+                case FLOAT:
+                    jsonBuilder.append("\"" + row.getFloat(col.getName()) + "\"");
+                    break;
+                case DOUBLE:
+                    jsonBuilder.append("\"" + row.getDouble(col.getName()) + "\"");
+                    break;
+                case UNIXTIME_MICROS:
+                    jsonBuilder.append("\"" + row.getLong(col.getName()) + "\"");
+                    break;
+                case BINARY:
+                    jsonBuilder.append("\"0x" + Hex.encodeHexString(row.getBinaryCopy(col.getName())) + "\"");
+                    break;
+                default:
+                    break;
+            }
+            if(columns.hasNext())
+                jsonBuilder.append(",");
+        }
+        jsonBuilder.append("}");
+        return jsonBuilder.toString();
+    }
+
+    protected Object parseValue(String value, ColumnSchema columnSchema) {
+        Object  parsedValue;
+        Type type = columnSchema.getType();
+
+        switch (type) {
+            case STRING:
+                if (value.isEmpty()) {
+                    throw new IllegalStateException(String.format("give value is Empty for type %s", type));
+                }
+                parsedValue = value;
+                break;
+            case INT8:
+                parsedValue = Integer.valueOf(value);
+                break;
+            case INT16:
+                parsedValue = Integer.valueOf(value);
+                break;
+            case INT32:
+                parsedValue = Integer.valueOf(value);
+                break;
+            case INT64:
+                parsedValue = Long.valueOf(value);
+                break;
+            case BOOL:
+                parsedValue = Boolean.valueOf(value);
+                break;
+            case FLOAT:
+                parsedValue = Float.valueOf(value);
+                break;
+            case DOUBLE:
+                parsedValue = Double.valueOf(value);
+                break;
+            case DECIMAL:
+                parsedValue = new BigDecimal(value);
+                break;
+            case UNIXTIME_MICROS:
+                parsedValue = Long.valueOf(value);
+                break;
+            case BINARY:
+                parsedValue = Byte.valueOf(value);
```

Comment:
This should return a byte `array`, not a byte.
Not sure in what format the user should be able to provide a value here but to be consistent with the JSON output, imo we should expect a hexadecimal.

So the implementation could be:
```java
                try {
                    parsedValue = Hex.decodeHex(value);
                } catch (DecoderException e) {
                    throw new RuntimeException(e);
                }
```